java/java-impl-inspections/src/com/intellij/codeInspection/BulkFileAttributesReadInspection.java

```diff
@@ -251,8 +252,23 @@ public void applyFix(@NotNull Project project, @NotNull ProblemDescriptor descri
                                                                    @NotNull String varName,
                                                                    @NotNull PsiType type,
                                                                    @Nullable PsiExpression initializer) {
+      final InspectionProjectProfileManager profileManager = InspectionProjectProfileManager.getInstance(parent.getProject());
+      final InspectionProfile inspectionProfile = profileManager.getCurrentProfile();
       PsiElementFactory elementFactory = PsiElementFactory.getInstance(parent.getProject());
-      PsiDeclarationStatement declaration = elementFactory.createVariableDeclarationStatement(varName, type, initializer);
+      PsiType displayType;
+      if (
+        initializer != null
+        && parent.getContext() != null
+        && PsiUtil.isLanguageLevel10OrHigher(parent.getContainingFile())
+        && (inspectionProfile.getInspectionTool(new VariableTypeCanBeExplicitInspection().getShortName(), parent.getProject()) == null
+            || inspectionProfile.getInspectionTool(new RedundantExplicitVariableTypeInspection().getShortName(), parent.getProject()) != null)
```

Comment:
Please respect a code style setting, rather than checking the inspection profile state. See the comment in the issue.